### PR TITLE
Replace rest-client in favor of faraday & update rack-attack (security issues #337)

### PR DESCRIPTION
**Pull request by vpetrusenko**
related to issue #337

### CODE REVIEW

**Comment by vpetrusenko** [comment]
Updating ruby-advisory-db ...
From https://github.com/rubysec/ruby-advisory-db
 * branch            master     -> FETCH_HEAD
Already up to date.
Updated ruby-advisory-db
ruby-advisory-db: 293 advisories
No vulnerabilities found

**Comment by yivo** [comment]
@vpetrusenko Please check that BTC & XRP deposits / trading / withdraws work. Also pls check API is able to limit requests by setting small limit and issuing multiple requests.

**Comment by yivo** [comment]
@vpetrusenko Any feedback?

**Comment by vpetrusenko** [comment]
@yivo deposits / trading / withdraws works correctly, now i try to check API limit

File: Gemfile [hunk 1]
```diff
@@ -57,10 +57,10 @@ gem 'grape', '~> 1.0.1'
 gem 'grape-entity', '~> 0.5.2'
 gem 'grape-swagger', '~> 0.27.3'
 gem 'grape-swagger-ui', '~> 2.2.8'
-gem 'rack-attack', '~> 3.0.0'
+gem 'rack-attack', '~> 4.3.1'
 gem 'easy_table'
 gem 'carrierwave', '~> 0.10.0'
-gem 'rest-client', '~> 1.6.8'
+gem 'faraday', '~> 0.12.2'
```
Comment (yivo) [inline Gemfile]: @vpetrusenko I think we don't specify patch level in version for faraday.

**Review by yivo — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 1 above)
Comment (vpetrusenko) [reply Gemfile]: @yivo okay

**Comment by vpetrusenko** [comment]
@yivo deposits / trading / withdraws works correctly, rack-attack also seems to works correctly, bundle audit cannot find any vulnerabilities

File: Gemfile [hunk 2]
```diff
@@ -57,10 +57,10 @@ gem 'grape', '~> 1.0.1'
 gem 'grape-entity', '~> 0.5.2'
 gem 'grape-swagger', '~> 0.27.3'
 gem 'grape-swagger-ui', '~> 2.2.8'
-gem 'rack-attack', '~> 3.0.0'
+gem 'rack-attack', '~> 4.3.1'
 gem 'easy_table'
 gem 'carrierwave', '~> 0.10.0'
-gem 'rest-client', '~> 1.6.8'
+gem 'faraday'
```
Comment (yivo) [inline Gemfile]: @vpetrusenko You should specify version range.

File: lib/coin_rpc/xrp.rb [hunk 3]
```diff
@@ -118,17 +118,17 @@ def sendtoaddress(address, amount, fee)
       issuer = fs.member.payment_addresses.find_by(currency: fs.currency_value)
 
       resp = JSON.parse(
-        RestClient.get(
+          Faraday.get(
```
Comment (yivo) [inline lib/coin_rpc/xrp.rb]: @vpetrusenko Why you have lot a leading spaced for each `Faraday` statement?

**Review by yivo — CHANGES_REQUESTED** [review]

**Comment by vpetrusenko** [comment]
@yivo made changes according to last review

**Review by yivo — APPROVED** [review]

**Pull request merged**